SensorCommunication/Sensor.cpp

```diff
@@ -0,0 +1,169 @@
+//
+// Created by pbrink on 22/03/18.
+//
+
+#include <stdexcept>
+#include <iostream>
+#include <cstring>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include "Sensor.h"
+
+Sensor::Sensor(string name,
+               string address,
+               unsigned long int samplingFrequency,
+               ReadWrite action,
+               string inputFileName,
+               string outputFileName) :
+                name(name), address(address), action(action)
+{
+    // ensure that input and output filenames are properly specified when necessary
+    if (action == ReadWrite::READ || action == ReadWrite::READ_WRITE) {
+        if (isFilenameAllowed(outputFileName)) {
+            this->outputFileName = outputFileName;
+            if (!createNamedPipe(outputFileName, true)) {
+                throw std::invalid_argument("Invalid output filename provided");
+            }
+        } else {
+            throw std::invalid_argument("Invalid output filename provided");
+        }
+    }
+    if (action == ReadWrite::WRITE || action == ReadWrite::READ_WRITE) {
+        if (isFilenameAllowed(inputFileName)) {
+            this->inputFileName = inputFileName;
+            if (!createNamedPipe(inputFileName, false)) {
+                throw std::invalid_argument("Invalid input filename provided");
+            }
+        } else {
+            throw std::invalid_argument("Invalid input filename provided");
+        }
+    }
+
+    // validate sampling frequency
+    if (samplingFrequency < SAMPLING_FREQUENCY_LOWER_LIMIT || samplingFrequency > SAMPLING_FREQUENCY_UPPER_LIMIT) {
+        throw std::invalid_argument("Invalid sampling frequency provided");
+    } else {
+        this->samplingFrequency = samplingFrequency;
+    }
+
+    // create the callback timer
+    timer = new CallBackTimer();
+}
+
+bool Sensor::isFilenameAllowed(string filename) {
+    // TODO: verification of good filenames and paths
+    return true;
+}
+
+bool Sensor::setSamplingFrequency(unsigned long int samplingFrequency) {
+    if (samplingFrequency < SAMPLING_FREQUENCY_LOWER_LIMIT || samplingFrequency > SAMPLING_FREQUENCY_UPPER_LIMIT) {
+        return false;
+    } else {
+        this->samplingFrequency = samplingFrequency;
+    }
+}
+
+bool Sensor::startReading() {
```

Comment:
since I commented on one of the older commits I wanna make sure that this isn't missed:
My personal opinion is that it would make more sense if the methods `startReading` and `stopReading` returned `true` if they successfully started or stopped the sensor. Since this doesn't really affect the functionality of the code though it isn't a real issue but I just thought I'd bring it up for discussion's sake. :eyes:

Reply:
Yeah, you're probably right, that would make more sense. I'll make that change